### DISCUSSION TITLE
ParticleOrbitalState - advice on inheriting from ParticleState

### DISCUSSION THREAD

**Discussion by jmbarr** [dstl/Stone-Soup#421] [Q&A]
I'm trying to create a new class from `ParticleState` and `OrbitalState`. I found this almost hilariously easy when combining `GaussianState` with `OrbitalState` to run derivations of Kalman filters on orbital scenarios.

I've come somewhat unstuck when trying to to the same thing with particle states though. Because the parameters don't match, `ParticleState` initiates on a list of `Particle`s (weighted state vectors), `OrbitalState` on a `StateVector`, I get an error in `super.__init__()`. Before I go down a rabbit hole for a solution, does anyone have an idea on how this might be solved?

**Comment by sdhiscocks** [comment]
Maybe something like the `Update` types, where you have a generic Astro and then construct State based types from it: https://github.com/dstl/Stone-Soup/blob/ee2d6834ce144163d03332e9452de24af81bd5c5/stonesoup/types/update.py#L54-L109
